src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts

```diff
@@ -161,18 +182,61 @@ declare const acquireVsCodeApi: any;
       line.appendChild(code);
     }
   }
-  function getCurrentSeverity() {
-    const stringMap = {
+
+  /**
+   * Transforms a severity string from a `Suggestion` object into a `CurrentSeverity` object.
+   *
+   * This function maps a severity string ('Low', 'Medium', 'High') to its corresponding
+   * numeric value. If the provided severity string is not one of the allowed values, the function
+   * returns `undefined`.
+   *
+   * @param {Suggestion['severity']} severity - The severity string to be mapped.
+   * @returns {CurrentSeverity | undefined} The mapped severity object, or undefined
+   */
+  function getCurrentSeverity(severity: Suggestion['severity']): CurrentSeverity | undefined {
+    const severityMap = {
       Low: 1,
       Medium: 2,
       High: 3,
     };
-    return suggestion
-      ? {
-          value: stringMap[suggestion.severity],
-          text: suggestion.severity,
-        }
-      : undefined;
+
+    const severityAllowedValues = Object.keys(severityMap);
+    if (!severityAllowedValues.includes(severity)) {
+      return undefined;
+    }
+
+    return {
+      value: severityMap[severity],
+      text: severity,
+    };
+  }
```

Comment:
Refactored this function since the check `suggestion ? {} : undefined` didn't guarantee that suggestion contained a `severity` property or that `severity` contained valid values to match `severityMap[suggestion.severity]`.